grove/tomography/tomography.py

```diff
@@ -0,0 +1,727 @@
+        process.
+
+        :param (qutip.Qobj|matrix-like) other: A unitary operator that induces a process as
+         ``rho -> other*rho*other.dag()``, can also be a superoperator or Pauli-transfer matrix.
+        :return: The process fidelity, a real number between 0 and 1.
+        :rtype: float
+        """
+        if isinstance(other, qt.Qobj):
+            if not other.issuper or other.superrep != SUPER:
+                sother = qt.to_super(other)
+            else:
+                sother = other
+            tm_other = self.pauli_basis.transfer_matrix(sother)
+        else:
+            tm_other = csr_matrix(other)
+        dimension = self.pauli_basis.ops[0].shape[0]
+        return np.trace(tm_other.T * self.r_est).real / dimension ** 2
+
+    def avg_gate_fidelity(self, other):
+        """
+        Compute the average gate fidelity of the estimated state with respect to a unitary process.
+        See `Chow et al., 2012, <https://doi.org/10.1103/PhysRevLett.109.060501>`_
+
+        :param (qutip.Qobj|matrix-like) other: A unitary operator that induces a process as
+            `rho -> other*rho*other.dag()`, alternatively a superoperator or Pauli-transfer matrix.
+        :return: The average gate fidelity, a real number between 1/(d+1) and 1, where d is the
+        Hilbert space dimension.
+        :rtype: float
+        """
+        process_fidelity = self.process_fidelity(other)
+        dimension = self.pauli_basis.ops[0].shape[0]
+        return (dimension * process_fidelity + 1.0) / (dimension + 1.0)
+
+    def to_super(self):
+        """
+        Compute the standard superoperator representation of the estimated process.
+
+        :return: The process as a superoperator.
+        :rytpe: qutip.Qobj
+        """
+        return self.sop
+
+    def to_choi(self):
+        """
+        Compute the choi matrix representation of the estimated process.
+
+        :return: The process as a choi-matrix.
+        :rytpe: qutip.Qobj
+        """
+        return qt.to_choi(self.sop)
+
+    def to_chi(self):
+        """
+        Compute the chi process matrix representation of the estimated process.
+
+        :return: The process as a chi-matrix.
+        :rytpe: qutip.Qobj
+        """
+        return qt.to_chi(self.sop)
+
+    def plot_pauli_transfer_matrix(self, ax):
+        """
+        Plot the elements of the Pauli transfer matrix.
+
+        :param matplotlib.Axes ax: A matplotlib Axes object to plot into.
+        """
+        title = "Estimated process"
+        ut.plot_pauli_transfer_matrix(self.r_est, ax, self.pauli_basis.labels, title)
+
+    def plot(self):
+        """
+        Visualize the process.
+
+        :return: The generated figure.
+        :rtype: matplotlib.Figure
+        """
+        fig, (ax1) = plt.subplots(1, 1, figsize=(10, 8))
+        self.plot_pauli_transfer_matrix(ax1)
+        return fig
+
+
+def bitstring_to_int(bitstring):
```

Comment:
you might want to consider using the python internal function `int(bitstring, 2)`

Reply:
:+1:

Reply:
Oh I see - this is actually intended to be use for lists of bits.